components/motor/ulnstepper/28byj-48.go

```diff
@@ -0,0 +1,478 @@
+// Package ulnstepper implements a GPIO based stepper motor with uln2003 controler.
+package ulnstepper
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/edaniels/golog"
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+	"go.viam.com/utils"
+
+	"go.viam.com/rdk/components/board"
+	"go.viam.com/rdk/components/generic"
+	"go.viam.com/rdk/components/motor"
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/operation"
+	"go.viam.com/rdk/registry"
+	"go.viam.com/rdk/resource"
+	rdkutils "go.viam.com/rdk/utils"
+)
+
+var (
+	model                = resource.NewDefaultModel("28byj48")
+	minStepDelay         = 0.002                  // minimum sleep time between steps
+	minDelayBetweenTicks = 100 * time.Microsecond // minimum sleep time between each ticks
+)
+
+// stepSequence contains high and low signal for uln2003 pins.
+// Treversing through stepSequence once is one step.
+var stepSequence = [8][4]bool{
+	{true, false, false, true},
+	{true, false, false, false},
+	{true, true, false, false},
+	{false, true, false, false},
+	{false, true, true, false},
+	{false, false, true, false},
+	{false, false, true, true},
+	{false, false, false, true},
+}
+
+// PinConfig defines the mapping of where motor are wired.
+type PinConfig struct {
+	In1 string `json:"In1"`
+	In2 string `json:"In2"`
+	In3 string `json:"In3"`
+	In4 string `json:"In4"`
+}
+
+// Config describes the configuration of a motor.
+type Config struct {
+	Pins              PinConfig `json:"pins"`
+	BoardName         string    `json:"board"`
+	RotationPerMinute float64   `json:"rotation_per_minute"`
+	TicksPerRotation  int       `json:"ticks_per_rotation"`
+}
+
+// Validate ensures all parts of the config are valid.
+func (config *Config) Validate(path string) ([]string, error) {
+	var deps []string
+	if config.BoardName == "" {
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "board")
+	}
```

Comment:
[minor], we need to check that all four pins are present in the config for this stepper motor model.